### PR TITLE
First run fallback folder

### PR DESCRIPTION
**Pull request by livarcocc**
Populating a nuget fallback folder during first run, instead of the nuget cache.

@dotnet/dotnet-cli 

Fixes https://github.com/dotnet/cli/issues/6100

### CODE REVIEW

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x64 Release Build

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x64 Release Build

File: src/Microsoft.DotNet.Configurer/CLIFallbackFolderPathCalculator.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.PlatformAbstractions;
+using NuGet.Common;
+
+namespace Microsoft.DotNet.Configurer
+{
+    public class CLIFallbackFolderPathCalculator
```
Comment (eerhardt) [inline src/Microsoft.DotNet.Configurer/CLIFallbackFolderPathCalculator.cs]: `CLI` should be `Cli`.  Everywhere.  It is a 3 letter acronym, so only the first letter should be capitalized.

See `Microsoft.DotNet.Cli.Utils`.

File: src/Microsoft.DotNet.Configurer/CLIFallbackFolderPathCalculator.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            get
+            {
+                string profileDir = Environment.GetEnvironmentVariable(
+                    RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "USERPROFILE" : "HOME");
+
+                return Path.Combine(profileDir, ".dotnet", "NuGetFallbackFolder");
+            }
+        }
+
+        public string NuGetUserSettingsDirectory
```
Comment (nguerrera) [inline src/Microsoft.DotNet.Configurer/CLIFallbackFolderPathCalculator.cs]: Consider making this expression bodied.

File: src/Microsoft.DotNet.Configurer/NuGetCacheSentinel.cs [hunk 3]
```diff
@@ -74,9 +62,9 @@ private void SetInProgressSentinel()
         {
             try
             {
-                if(!Directory.Exists(NuGetCachePath))
+                if(!Directory.Exists(_nugetCachePath))
```
Comment (nguerrera) [inline src/Microsoft.DotNet.Configurer/NuGetCacheSentinel.cs]: nit: missing space after `if`

File: src/Microsoft.DotNet.Configurer/NuGetPackagesArchiver.cs [hunk 4]
```diff
@@ -10,33 +10,15 @@ namespace Microsoft.DotNet.Configurer
...
-        {
-        }
-
-        internal NuGetPackagesArchiver(IDirectory directory)
-        {
-            _temporaryDirectory = directory.CreateTemporaryDirectory();
-        }
-
-        public string ExtractArchive()
+        public void ExtractArchive(string archiveDestination)
```
Comment (nguerrera) [inline src/Microsoft.DotNet.Configurer/NuGetPackagesArchiver.cs]: Lots of public API changes. I know I've seen other PRs where changing Microsoft.DotNet.* public API was turned away. How can I tell which assemblies are OK to break like this?

Comment (livarcocc) [inline src/Microsoft.DotNet.Configurer/NuGetPackagesArchiver.cs]: This is not published as a nupkg. The only one that is is Cli.Utils.

**Comment by livarcocc** [comment]
@dotnet-bot Test Ubuntu16.10 x64 Debug Build

**Pull request merged**